org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/services/IEditorServicesStrategy.java

```diff
@@ -34,4 +36,14 @@
      */
     FailOrSuccessResult<StyleResult, IResult> highlight(String source);
 
+    /**
+     * Attempts to provide pretty printing of the input.
+     *
+     * @param input
+     *            the term that must be pretty printed.
+     * @return {@link FailOrSuccessResult} - A {@link PrintResult} containing a human readable
+     *         text plus fold regions, or a failed result.
+     */
+    FailOrSuccessResult<PrintResult, IResult> print(ISpoofaxTermResult<?> input);
```

Comment:
This also provides folding right? Maybe `foldAndPrint` or simply `fold` is more appropriate (result is a textual (pretty-printed) representation of a folding).